data-protocols/dsp/dsp-core/src/main/java/org/eclipse/edc/protocol/dsp/DspError.java

```diff
@@ -0,0 +1,45 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp;
+
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import org.eclipse.edc.jsonld.spi.JsonLdKeywords;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static org.eclipse.edc.protocol.dsp.type.DspPropertyAndTypeNames.DSPACE_PROCESS_ID;
+import static org.eclipse.edc.protocol.dsp.type.DspPropertyAndTypeNames.DSPACE_PROPERTY_CODE;
+import static org.eclipse.edc.protocol.dsp.type.DspPropertyAndTypeNames.DSPACE_PROPERTY_REASON;
+
+public final class DspError {
+
+    public static JsonObject create(String type, Optional<String> processId, String code, String... messages) {
```

Comment:
Per my previous comment, this should use a builder. It will add parameter safety, remove the need for optional, and make passing messages fluent.

Error will also need to synthesize a @context and set @vocab to the EDC namsespace.

Reply:
I will create a builder.